test/math.ts

```diff
@@ -53,4 +53,103 @@ test("Should be able to use the cos function with degrees and radians", (assert)
     ~~~
   `);
   assert.end();
-})
\ No newline at end of file
+})
+
+test("Test that string concatenation is still working after NaN change.", (assert) => {
+  let expected = {
+    insert: [
+      ["a", "tag", "div"],
+      ["a", "text", "Test Testy"],
+    ],
+    remove: [],
+  };
+
+  evaluate(assert, expected, `
+    ~~~
+    search
+      a = "Test "
+      b = "Testy"
+      x =  a + b
+    bind @browser
+    [#div text: x]
+    ~~~
+  `);
+  assert.end();
+});
+
+test("Divide by zero should return nothing.", (assert) => {
+  let expected = {
+    insert: [],
+    remove: [],
+  };
+
+  evaluate(assert, expected, `
+    Now consider this:
+    ~~~
+    search
+      x = 1 / 0
+    bind @browser
+      [#div text:x]
+    ~~~
+  `);
+  assert.end();
+});
+
+test("Divide by zero in an if statement should be detectable.", (assert) => {
+  let expected = {
+    insert: [
+      ["a", "tag", "div"],
+      ["a", "text", "Divide by zero"],
+    ],
+    remove: [],
+  };
+
+  evaluate(assert, expected, `
+    ~~~
+    search
+      a = 1
+      b = 0
+      x = if a / b then "Ooops"
+          else "Divide by zero"
+    bind @browser
+    [#div text: x]
+    ~~~
+  `);
+  assert.end();
+});
+
+test("ACosh < 1 should return nothing.", (assert) => {
+  let expected = {
+    insert: [],
+    remove: [],
+  };
+
+  evaluate(assert, expected, `
+    Now consider this:
+    ~~~
+    search
+      x = acosh[value: 0.999999999999999]
+    bind @browser
+      [#div text:x]
+    ~~~
+  `);
+  assert.end();
+});
+
+test("ATanH < -1 and > 1 should return nothing.", (assert) => {
+  let expected = {
+    insert: [],
+    remove: [],
+  };
```

Comment:
Yeah, the behavior you have is correct here. What we want to verify is that if anything would otherwise return NaN that it instead returns nothing. It's not an error, it's the absence of a result.